src/arrayMethodJoin.js

```diff
@@ -3,9 +3,37 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+    }
+    let str = '';
+    let middleString = '';
```

Comment:
`str` and `middleString` are not really good names, I don't see the difference between them from their names